### PR TITLE
Update Kusto Go SDK version , fixes issue #22771

### PR DESCRIPTION
**Pull request by ag-ramachandran**
**Description:** 
The underlying SDK has an update to fix an issue where the network connections are flaky. To get the metadata to connect to the service , a HTTP metadata call is issued. If this call fails , it is not retried and the init is interrupted till restart. The underlying SDK fixes it by reset of the lock that was causing this behavior

**Link to tracking Issue:** 
#22771

**Testing:** 

* Unit and E2E tests run
`--- PASS: Test_mapToAdxMetric (0.01s)
    --- PASS: Test_mapToAdxMetric/counter_over_time (0.00s)
    --- PASS: Test_mapToAdxMetric/int_counter_over_time (0.00s)
    --- PASS: Test_mapToAdxMetric/nil_counter_over_time (0.00s)
    --- PASS: Test_mapToAdxMetric/simple_histogram_with_value (0.00s)
    --- PASS: Test_mapToAdxMetric/nil_gauge_value (0.00s)
    --- PASS: Test_mapToAdxMetric/int_gauge_value (0.00s)
    --- PASS: Test_mapToAdxMetric/float_gauge_value (0.00s)
    --- PASS: Test_mapToAdxMetric/summary (0.00s)
    --- PASS: Test_mapToAdxMetric/nil_summary (0.00s)
    --- PASS: Test_mapToAdxMetric/unknown_type (0.00s)
=== RUN   TestLoadConfig/azuredataexplorer/2
=== RUN   TestLoadConfig/azuredataexplorer/3
--- PASS: TestLoadConfig (0.02s)
    --- PASS: TestLoadConfig/azuredataexplorer (0.01s)
    --- PASS: TestLoadConfig/azuredataexplorer/2 (0.00s)
    --- PASS: TestLoadConfig/azuredataexplorer/3 (0.00s)
--- PASS: TestCreateTracesExporterE2E (9.25s)
--- PASS: TestCreateLogsExporterE2E (9.37s)
--- PASS: TestCreateMetricsExporterE2E (10.36s)
PASS
ok      github.com/open-telemetry/opentelemetry-collector-contrib/exporter/azuredataexplorerexporter    31.642s`

**Documentation:**  
No changes

### CODE REVIEW

**Review by ben-childs-docusign — COMMENTED** [review]
Thank you for the fix!

**Comment by mx-psi** [comment]
Can you fix the merge conflicts? Thanks!

**Comment by ag-ramachandran** [comment]
> Can you fix the merge conflicts? Thanks!

G'day @mx-psi, merged and updated!

File: exporter/azuredataexplorerexporter/e2e_test.go [hunk 1]
```diff
@@ -235,14 +231,7 @@ func TestCreateMetricsExporterE2E(t *testing.T) {
...
 
-func prepareQuery(tID string) (kusto.ParamTypes, kusto.QueryValues) {
-	kustoDefinitions := make(kusto.ParamTypes)
-	kustoParameters := make(kusto.QueryValues)
-	kustoDefinitions["TID"] = kusto.ParamType{Type: types.String}
-	kustoParameters["TID"] = tID
-	return kustoDefinitions, kustoParameters
-}
-
+// prepareQuery no longer used
```
Comment (dmitryax) [inline exporter/azuredataexplorerexporter/e2e_test.go:234]: This seems like a review comment, not a code comment. Please remove

Comment (ag-ramachandran) [reply exporter/azuredataexplorerexporter/e2e_test.go]: Hello @dmitryax , Thanks for the comment. Fixed this and pushed the changes

**Comment by ag-ramachandran** [comment]
Hello @mx-psi,@dmitryax. Sorry for 1 more message. When you get sometime could you please review/approve merge

**Review by dmitryax — APPROVED** [review]

**Pull request merged**